src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -240,18 +243,30 @@ public void drive(double xSpeed, double ySpeed, double rot, boolean fieldRelativ
 
       xSpeedCommanded = m_currentTranslationMag * Math.cos(m_currentTranslationDir);
       ySpeedCommanded = m_currentTranslationMag * Math.sin(m_currentTranslationDir);
-      m_currentRotation = m_rotLimiter.calculate(rot);
+      m_rotationCommanded = m_rotLimiter.calculate(rot);
 
     } else {
       xSpeedCommanded = xSpeed;
       ySpeedCommanded = ySpeed;
-      m_currentRotation = rot;
+      m_rotationCommanded = rot;
     }
 
     // Convert the commanded speeds into the correct units for the drivetrain
     double xSpeedDelivered = xSpeedCommanded * DriveConstants.kMaxSpeedMetersPerSecond;
     double ySpeedDelivered = ySpeedCommanded * DriveConstants.kMaxSpeedMetersPerSecond;
-    double rotDelivered = m_currentRotation * DriveConstants.kMaxAngularSpeed;
+    double rotDelivered = 0.0;
+    if (DriverStation.isTeleopEnabled()) {
+      intendedRotation = intendedRotation
+          .rotateBy(new Rotation2d((-m_rotationCommanded) * ((timeOfLastLoop - System.currentTimeMillis()) / 1000)
+              * DriveConstants.kMaxAngularSpeed));
+      timeOfLastLoop = System.currentTimeMillis();
+      double Kp = 0.1; // P gain (may be tuned)
+      double error = intendedRotation.minus(getHeading()).getDegrees(); // Calculate error
```

Comment:
The calculations here are done in radians, so converting the error to degrees isn't necessary.  Leaving the intended rotation value in radians would accomplish the same  logic, although you may need to reduce your initial value of `Kp` to see the same behavior.